src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -295,13 +295,19 @@ private static unsafe string Ctor(ReadOnlySpan<char> value)
 
         public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)
         {
-            ArgumentNullException.ThrowIfNull(action);
+            if (action is null)
```

Comment:
@stephentoub: Just out of curiosity, why was this changed?

Not related to this PR: in `Ctor(char c, int count)` the count is checked twice:
https://github.com/stephentoub/runtime/blob/3298c130f2540f252647cfe0129ed93fb0affcf1/src/libraries/System.Private.CoreLib/src/System/String.cs#L268-L272

Better to use `ThrowHelper.ThrowArgumentOutOfRangeException(...)` there?

Reply:
This pattern results in ever so slightly smaller asm, and as it's in a generic method and we already had the helper and the appropriately named constant argument, it was trivial to switch.